src/components/OptionListContextProvider.tsx

```diff
@@ -64,22 +64,11 @@ function OptionsListContextProvider({reports, children}: OptionsListProviderProp
             return;
         }
 
-        const lastUpdatedReport = ReportUtils.getLastUpdatedReport();
-
-        if (!lastUpdatedReport) {
-            return;
-        }
-
-        const newOption = OptionsListUtils.createOptionFromReport(lastUpdatedReport, personalDetails);
-        const replaceIndex = options.reports.findIndex((option) => option.reportID === lastUpdatedReport.reportID);
-
-        if (replaceIndex === -1) {
-            return;
-        }
+        const newReports = OptionsListUtils.createOptionList({}, reports).reports;
```

Comment:
Generating the whole optionList whenever any report changes looks inefficient. Can we just update the last three or five updated reports?

Reply:
I am sorry. I see that there can be many policy expense chats of other invited members as well. So just deleting 5 reports might not suffice because the other reports still appear in the results.

https://github.com/Expensify/App/assets/102477862/3894b75b-1b32-42a7-8917-16ddd8ea25e2